src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -1081,6 +1081,11 @@ internal static Task LogDowngradeWarningsOrErrorsAsync(IEnumerable<RestoreTarget
             else
             {
                 _success = false;
+                var frameworkRuntimePairs = CreateFrameworkRuntimePairs(_request.Project, RequestRuntimeUtility.GetRestoreRuntimes(_request));
```

Comment:
Shouldn't we have explicit types here and in the `foreach`?  Also, is it worth adding a comment here or would a `git blame` be enough for people to understand why this is needed (not sure if anyone would ever wonder about it so its up to you)

Reply:
I'll change it to use explicit types. 
I can add a small comment, you're right it'll help the reader a lot.